qutebrowser/completion/completer.py

```diff
@@ -151,6 +151,19 @@ def _get_completion_model(self, completion, parts, cursor_part):
             except KeyError:
                 # No completion model for this section/option.
                 model = None
+        elif completion == usertypes.Completion.keybinding:
+            # look for --mode to provide mode-specific binding completions
+            # since flags are ignored by most completers, we need to re-parse
+            parts, cursor_part = self._filter_cmdline_parts(self.split(),
+                                                            cursor_part,
+                                                            ignore_flags=False)
+            if parts[cursor_part - 1].startswith("--mode="):
```

Comment:
So, it turns out `parse_known_args` allows _extra_ args, but not _missing_ args. So either `parse_args` or `parse_known_args` will fail here as we don't have `command` yet (that's what we want to complete!). This can be fixed by making `command` optional -- I was thinking of having `bind <key>` print the current binding of `key` (similar to `map` in vim).

However, the more complicated issue is that we have no way to distinguish between flags that do and don't take args. That means that given `--mode caret`, `cursor_part` will be offset by 1 (as `caret` looks like a positional arg) and it is hard to determine when we should/shouldn't show completions. This applies even if we just show all completions without trying to filter by mode. `bind --mode caret` will start showing completions for the `key` rather than the `command`.